build.rs

```diff
@@ -155,6 +155,20 @@ fn main() {
 
     ccfg
         .flag("-Wall")
+        .flag("-Wno-pointer-sign")
+        .flag("-Wno-unused-but-set-variable")
+        .flag("-Wno-unused-function")
+        .flag("-Wno-unused-value")
+        .flag("-Wno-unused-variable")
+        .flag("-Wmissing-include-dirs")
+        .flag("-Wredundant-decls")
+        .flag("-Wpointer-arith")
+        .flag("-Wformat=2")
+        .flag("-Wundef")
+        .flag("-Wunreachable-code")
+        .flag("-Wstrict-prototypes")
+        .flag("-Wold-style-definition")
+        .flag("-Wwrite-strings")
```

Comment:
I am concerned that these options could cause problems for platforms where the compiler is not actually `gcc`. (While the Rust crate is named `gcc`, it will use whatever system C compiler it thinks best.) Is there a way to test whether the compiler is `gcc` and only add these flags if we can confirm that it is? If not, I would rather not apply all of these extra flags.